### PR TITLE
feat: add QRCodeDetector in models.artefacts

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR adds a QRCodeDetector in artefacts, based on cv2 QRCodeDetector, which detects and decode QR codes on images
Samples are displayed below.
Any feedback is welcome!

qr1
qr3
qr4
qr4
qr5
qr6

### CODE REVIEW

**Comment by charlesmindee** [comment]
It takes 10s to run this detector 100 times on the same image, a 1024 x 1024 picture with a QR code on it. Experiments were conducted on a CPU (intel core i7 9th gen).

File: doctr/models/artefacts/qrcode.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class QRCodeDetector:
+
+    """ Implements a QR-code detector.
+    Based on open CV QRCodeDetector. Returns both localization and decoded text.
+    """
+
+    def __init__(
+        self,
+    ) -> None:
+        self.detector = cv2.QRCodeDetector()
```
Comment (fg-mindee) [inline doctr/models/artefacts/qrcode.py:22]: just need to inherit from cv2.QRCodeDetector, so that you don't have to implement the constructor ;)

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! Small improvement suggestion in the comments

**Review by fg-mindee — COMMENTED** [review]
there seems to be a segmentation fault in the unittest

**Comment by fg-mindee** [comment]
Small hypothesis here: cv2 is using a DL model under the hood or something similarly heavy on computation resources :thinking: 
(Since OpenCV 4, I think there has been a lot of those high computation features)

**Comment by charlesmindee** [comment]
Since we can't fix the issue with the test, I am closing this for now

**Pull request closed without merging**